application/system_processor/decoder/LogType.py

```
@@ -21,7 +21,13 @@ def isFunction(self):
     
     def isUnique(self):
         return getattr(self, "isUnique")
+    
+    def getLineNo(self):
+        return getattr(self, "lineno")
         
     def getName(self):
-        return getattr(self, "name")
+        if hasattr(self, "name"):
+            return getattr(self, "name")
```

Ruff (0.8.2)

Do not call `getattr` with a constant attribute value. It is not any safer than normal property access.

Replace `getattr` with attribute access

(B009)